styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java

```diff
@@ -261,31 +263,51 @@ public void shouldExecuteRetryIfDelayIsReset() {
   }
 
   @Test
-  public void shouldExecuteNewTriggers() {
-    initWorkflow(workflowUsingResources(WORKFLOW_ID1));
+  public void shouldExecuteNewTriggersOlderPartitionFirst() {
+    var stateData = StateData.newBuilder().tries(0).build();
 
-    StateData stateData = StateData.newBuilder().tries(0).build();
+    for (var i = 1; i <= 5; i++) {
+      for (var j = 9; j >= 0; j--) { // enqueuing newer first, older last
+        var workflowId = WorkflowId.create("styx", "example" + i);
+        initWorkflow(workflowUsingResources(workflowId));
+        var instance = WorkflowInstance.create(workflowId, "2016-12-02T0" + j);
+        populateActiveStates(RunState.create(instance, State.QUEUED, stateData,
+            time.get().minus(j, ChronoUnit.SECONDS)));
+      }
+    }
 
-    populateActiveStates(RunState.create(INSTANCE_1, State.QUEUED, stateData, time.get()));
+    scheduler.tick();
 
-    List<WorkflowInstance> workflowInstances = new ArrayList<>();
+    var instancesOlderFirst = activeStates.keySet().stream()
+        .sorted(WorkflowInstance.KEY_COMPARATOR.thenComparing(WorkflowInstance::parameter));
```

Comment:
I think I would prefer duplicating the group by and then sorting logic than this double sorting.

Reply:
Hit enter too fast. To me the reason is to avoid doing the translation between group by + linked hashmap + sorting + merging and double sorting when reading the code. But I see the simplification here too. So not a strong opinion.

Reply:
I didn't want to use the grouping because I felt that we were reusing the code from production code. But I will remove the ordering and create for loops with the verification inside to be explicit that we are testing that older goes first